.docker/run-flyway

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
+# Runs flyway and whatever provided command against the MISO DB that is set up with this container
+# Relies on environment variables: MISO_DB_PASS_FILE, MISO_DB_USER, MISO_DB_HOST_PORT, MISO_DB, MISO_FILES_DIR
+# the first argument given to this script is passed fully onto the flyway command
+# e.g.  
+#   run-flyway migrate : applies all of the migrations on the classpath and in /flyway/migrations
+#   run-flyway clean : rolls back changes
+
+password=$(cat ${MISO_DB_PASS_FILE})
+
+command="flyway/flyway  -user='${MISO_DB_USER}' -password='${password}' -url='jdbc:mysql://${MISO_DB_HOST_PORT}/${MISO_DB}?autoReconnect=true&zeroDateTimeBehavior=convertToNull&useUnicode=true&characterEncoding=UTF-8' -outOfOrder=true -locations=classpath:db/migration,classpath:uk.ac.bbsrc.tgac.miso.db.migration,filesystem:/flyway/migrations -placeholders.filesDir='${MISO_FILES_DIR}'"
+
+eval "${command} $1"
+
+rc=$?; if [[ $rc != 0 ]]
+then
+    eval "${command} clean"
```

Comment:
Clean doesn't roll back; it wipes the entire DB. I think that's probably not a good idea here. If a migration fails, it should be up to the person who ran it to clean it up how they want, which should usually mean restoring a backup

Reply:
fixed